From 2fec055d2286dde9c9ff1b67511ddeac04a781e7 Mon Sep 17 00:00:00 2001
From: Marek Vasut <marex@nabladev.com>
Date: Sat, 20 Jul 2024 15:31:48 +0200
Subject: [PATCH 1/2] etnaviv: hwdb: update gc_feature_database from ST

Updates the existing gc_feature_database.h file with changes from
https://github.com/STMicroelectronics/gcnano-binaries/blob/gcnano-6.4.21-binaries/gcnano-driver-stm32mp/hal/kernel/inc/gc_feature_database.h

git commit: 7c181cacf89f918039e64934fdc33fe817a052cd

Removed fields:
- NN_HIGH_PERF_DECODER

Upstream-Status: Backport [8237a65e94306d2beab0de0191d9af1c5970d350]
Reviewed-by: Christian Gmeiner <cgmeiner@igalia.com>
Signed-off-by: Marek Vasut <marex@nabladev.com>
---
 src/etnaviv/hwdb/st/gc_feature_database.h | 181 ++++++++++++++++++++--
 1 file changed, 171 insertions(+), 10 deletions(-)

diff --git a/src/etnaviv/hwdb/st/gc_feature_database.h b/src/etnaviv/hwdb/st/gc_feature_database.h
index fc2f75d0972..c3371022327 100644
--- a/src/etnaviv/hwdb/st/gc_feature_database.h
+++ b/src/etnaviv/hwdb/st/gc_feature_database.h
@@ -53,7 +53,7 @@
 *****************************************************************************/
 
 
-/*Auto created on 2024-07-26 17:55*/
+/*Auto created on 2024-12-23 15:24*/
 #ifndef _gc_feature_database_h_
 #define _gc_feature_database_h_
 
@@ -107,6 +107,8 @@ typedef struct
     gctUINT32 PS_INSTRUCTION_COUNT;
     gctUINT32 PS_INPUT_COMPONENTS;
     gctUINT32 AIGM_MAX_SIZE;
+    gctUINT32 GPU_AXI_BUS_WIDTH;
+    gctUINT32 NumDPPUsPerCluster;
     gctUINT32 NNMadPerCore;
     gctUINT32 NNCoreCount;
     gctUINT32 NN_ACTIVE_CORE_COUNT;
@@ -171,6 +173,7 @@ typedef struct
     gctUINT32 NN_COMMAND_BUFFER_SIZE;
     gctUINT32 DECOMP_VZ_GROUP_BITS;
     gctUINT32 NN_LOOP1_DP_NUMBER;
+    gctUINT32 NN_KERNEL_FIFO_IN_WIDTH;
     gctUINT32 REG_FastClear:1;
     gctUINT32 REG_SpecialAntiAliasing:1;
     gctUINT32 REG_Pipe3D:1;
@@ -604,6 +607,8 @@ typedef struct
     gctUINT32 DX11_SH_RCP_SQRT_PRECISSION:1;
     gctUINT32 TILED_RESOURCE:1;
     gctUINT32 SH_FP64:1;
+    gctUINT32 SH_INSTR_DENORM_SUPPORT:1;
+    gctUINT32 SH_GA_SUPPORT:1;
     gctUINT32 SH_SM6:1;
     gctUINT32 VGPU:1;
     gctUINT32 DEC_NANO:1;
@@ -805,7 +810,6 @@ typedef struct
     gctUINT32 NN_BF16_I4_I8_QUANTIZATION:1;
     gctUINT32 NN_HIGH_PERF_DECODE_SPLIT_STAGE:1;
     gctUINT32 NN_POST_PROCESSOR_FL32:1;
-    gctUINT32 NN_HIGH_PERF_DECODER:1;
     gctUINT32 NN_GROUP_QUANT_PHASE1:1;
     gctUINT32 NN_SH_IN_PARALLEL:1;
     gctUINT32 NN_ASYNC_DMA:1;
@@ -814,6 +818,21 @@ typedef struct
     gctUINT32 NN_SP_ENHANCEMENT:1;
     gctUINT32 NN_SUPPORT_SEPARATE_STREAMBUF_ADDR:1;
     gctUINT32 NN_TF32_MAC:1;
+    gctUINT32 NN_SP_FP28_PRECISION:1;
+    gctUINT32 NN_VIPSRAM_SHARE_AMONG_CLUSTERS:1;
+    gctUINT32 NN_SUPPORT_UNIQUE_ID:1;
+    gctUINT32 NN_FAST_VIPSRAM:1;
+    gctUINT32 NN_MOVE_DECOMPRESSOR_OUT_OF_CORE:1;
+    gctUINT32 NN_SPLITKZ_REDUCE_SUM:1;
+    gctUINT32 NN_SUPPORT_FFD:1;
+    gctUINT32 NN_SUPPORT_FP8_DOUBLE_MAC:1;
+    gctUINT32 NN_SUPPORT_DENORM:1;
+    gctUINT32 NN_SUPPORT_POST_OP_FUSE:1;
+    gctUINT32 T3D_SYNC_DMA:1;
+    gctUINT32 NN_GENERIC_DMA:1;
+    gctUINT32 NN_DOUBLE_CONV_OUT:1;
+    gctUINT32 CMD_LOOP_STATE_LOAD_SAVE:1;
+    gctUINT32 DYNAMIC_ACTIVATION_SCALE:1;
     gctUINT32 NN_PER3DTILE_BUBBLE_FIX:1;
     gctUINT32 NN_CACHELINE_MODE_PERF_FIX:1;
     gctUINT32 NN_CONV1x1_PERF_FIX:1;
@@ -952,6 +971,10 @@ typedef struct
     gctUINT32 B2B_RETRUN_NN_CMD_DONE_FIX:1;
     gctUINT32 MULTI_AXI_ID_VLW_3D_TILE_INFO_FIX:1;
     gctUINT32 TRSP2_SPECIAL_WORDSIZE_FIX:1;
+    gctUINT32 FIX_PRE_CORE_LOTS_ZEROSKIP_KERNEL_FURTHER_FIX:1;
+    gctUINT32 NN_2ND_IMAGE_TILE_LIMIT_1024B_FIX:1;
+    gctUINT32 MULTIAXIID_TBUF_STALL_READY_FIX:1;
+    gctUINT32 V940_TILESIZE_1024N_FIX:1;
     gctUINT32 NN_INTERLEVE8:1;
     gctUINT32 NN_FP16_ALU:1;
     gctUINT32 NN_INT16_ALU:1;
@@ -967,8 +990,8 @@ typedef struct
     gctUINT32 IMAGE_PARTIAL_CACHE:1;
 } gcsFEATURE_DATABASE;
 
-#define FEATURE_BIT_START 120
-#define FEATURE_BIT_END 913
+#define FEATURE_BIT_START 123
+#define FEATURE_BIT_END 936
 static gcsFEATURE_DATABASE gChipInfo[] = {
     /* gc7000nano_0x4652 */
     {
@@ -1019,6 +1042,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_PS_INSTRUCTION_COUNT */
         0x0, /* gcFEATURE_VALUE_PS_INPUT_COMPONENTS */
         0x0, /* gcFEATURE_VALUE_AIGM_MAX_SIZE */
+        0x0, /* gcFEATURE_VALUE_GPU_AXI_BUS_WIDTH */
+        0x0, /* gcFEATURE_VALUE_NumDPPUsPerCluster */
         0x0, /* gcFEATURE_VALUE_NNMadPerCore */
         0x0, /* gcFEATURE_VALUE_NNCoreCount */
         0x0, /* gcFEATURE_VALUE_NN_ACTIVE_CORE_COUNT */
@@ -1083,6 +1108,7 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_NN_COMMAND_BUFFER_SIZE */
         0x0, /* gcFEATURE_VALUE_DECOMP_VZ_GROUP_BITS */
         0x0, /* gcFEATURE_VALUE_NN_LOOP1_DP_NUMBER */
+        0x0, /* gcFEATURE_VALUE_NN_KERNEL_FIFO_IN_WIDTH */
         0x0, /* gcFEATURE_BIT_REG_FastClear */
         0x0, /* gcFEATURE_BIT_REG_SpecialAntiAliasing */
         0x1, /* gcFEATURE_BIT_REG_Pipe3D */
@@ -1516,6 +1542,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_DX11_SH_RCP_SQRT_PRECISSION */
         0x0, /* gcFEATURE_BIT_TILED_RESOURCE */
         0x0, /* gcFEATURE_BIT_SH_FP64 */
+        0x0, /* gcFEATURE_BIT_SH_INSTR_DENORM_SUPPORT */
+        0x0, /* gcFEATURE_BIT_SH_GA_SUPPORT */
         0x0, /* gcFEATURE_BIT_SH_SM6 */
         0x0, /* gcFEATURE_BIT_VGPU */
         0x0, /* gcFEATURE_BIT_DEC_NANO */
@@ -1717,7 +1745,6 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_BF16_I4_I8_QUANTIZATION */
         0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODE_SPLIT_STAGE */
         0x0, /* gcFEATURE_BIT_NN_POST_PROCESSOR_FL32 */
-        0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODER */
         0x0, /* gcFEATURE_BIT_NN_GROUP_QUANT_PHASE1 */
         0x0, /* gcFEATURE_BIT_NN_SH_IN_PARALLEL */
         0x0, /* gcFEATURE_BIT_NN_ASYNC_DMA */
@@ -1726,6 +1753,21 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_SP_ENHANCEMENT */
         0x0, /* gcFEATURE_BIT_NN_SUPPORT_SEPARATE_STREAMBUF_ADDR */
         0x0, /* gcFEATURE_BIT_NN_TF32_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SP_FP28_PRECISION */
+        0x0, /* gcFEATURE_BIT_NN_VIPSRAM_SHARE_AMONG_CLUSTERS */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_UNIQUE_ID */
+        0x0, /* gcFEATURE_BIT_NN_FAST_VIPSRAM */
+        0x0, /* gcFEATURE_BIT_NN_MOVE_DECOMPRESSOR_OUT_OF_CORE */
+        0x0, /* gcFEATURE_BIT_NN_SPLITKZ_REDUCE_SUM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FFD */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FP8_DOUBLE_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_DENORM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_POST_OP_FUSE */
+        0x0, /* gcFEATURE_BIT_T3D_SYNC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_GENERIC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_DOUBLE_CONV_OUT */
+        0x0, /* gcFEATURE_BIT_CMD_LOOP_STATE_LOAD_SAVE */
+        0x0, /* gcFEATURE_BIT_DYNAMIC_ACTIVATION_SCALE */
         0x0, /* gcFEATURE_BIT_NN_PER3DTILE_BUBBLE_FIX */
         0x0, /* gcFEATURE_BIT_NN_CACHELINE_MODE_PERF_FIX */
         0x0, /* gcFEATURE_BIT_NN_CONV1x1_PERF_FIX */
@@ -1864,6 +1906,10 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_B2B_RETRUN_NN_CMD_DONE_FIX */
         0x0, /* gcFEATURE_BIT_MULTI_AXI_ID_VLW_3D_TILE_INFO_FIX */
         0x0, /* gcFEATURE_BIT_TRSP2_SPECIAL_WORDSIZE_FIX */
+        0x0, /* gcFEATURE_BIT_FIX_PRE_CORE_LOTS_ZEROSKIP_KERNEL_FURTHER_FIX */
+        0x0, /* gcFEATURE_BIT_NN_2ND_IMAGE_TILE_LIMIT_1024B_FIX */
+        0x0, /* gcFEATURE_BIT_MULTIAXIID_TBUF_STALL_READY_FIX */
+        0x0, /* gcFEATURE_BIT_V940_TILESIZE_1024N_FIX */
         0x0, /* gcFEATURE_BIT_NN_INTERLEVE8 */
         0x0, /* gcFEATURE_BIT_NN_FP16_ALU */
         0x0, /* gcFEATURE_BIT_NN_INT16_ALU */
@@ -1927,6 +1973,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_PS_INSTRUCTION_COUNT */
         0x0, /* gcFEATURE_VALUE_PS_INPUT_COMPONENTS */
         0x0, /* gcFEATURE_VALUE_AIGM_MAX_SIZE */
+        0x0, /* gcFEATURE_VALUE_GPU_AXI_BUS_WIDTH */
+        0x0, /* gcFEATURE_VALUE_NumDPPUsPerCluster */
         0x0, /* gcFEATURE_VALUE_NNMadPerCore */
         0x0, /* gcFEATURE_VALUE_NNCoreCount */
         0x0, /* gcFEATURE_VALUE_NN_ACTIVE_CORE_COUNT */
@@ -1991,6 +2039,7 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_NN_COMMAND_BUFFER_SIZE */
         0x0, /* gcFEATURE_VALUE_DECOMP_VZ_GROUP_BITS */
         0x0, /* gcFEATURE_VALUE_NN_LOOP1_DP_NUMBER */
+        0x0, /* gcFEATURE_VALUE_NN_KERNEL_FIFO_IN_WIDTH */
         0x0, /* gcFEATURE_BIT_REG_FastClear */
         0x0, /* gcFEATURE_BIT_REG_SpecialAntiAliasing */
         0x1, /* gcFEATURE_BIT_REG_Pipe3D */
@@ -2424,6 +2473,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_DX11_SH_RCP_SQRT_PRECISSION */
         0x0, /* gcFEATURE_BIT_TILED_RESOURCE */
         0x0, /* gcFEATURE_BIT_SH_FP64 */
+        0x0, /* gcFEATURE_BIT_SH_INSTR_DENORM_SUPPORT */
+        0x0, /* gcFEATURE_BIT_SH_GA_SUPPORT */
         0x0, /* gcFEATURE_BIT_SH_SM6 */
         0x0, /* gcFEATURE_BIT_VGPU */
         0x0, /* gcFEATURE_BIT_DEC_NANO */
@@ -2625,7 +2676,6 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_BF16_I4_I8_QUANTIZATION */
         0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODE_SPLIT_STAGE */
         0x0, /* gcFEATURE_BIT_NN_POST_PROCESSOR_FL32 */
-        0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODER */
         0x0, /* gcFEATURE_BIT_NN_GROUP_QUANT_PHASE1 */
         0x0, /* gcFEATURE_BIT_NN_SH_IN_PARALLEL */
         0x0, /* gcFEATURE_BIT_NN_ASYNC_DMA */
@@ -2634,6 +2684,21 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_SP_ENHANCEMENT */
         0x0, /* gcFEATURE_BIT_NN_SUPPORT_SEPARATE_STREAMBUF_ADDR */
         0x0, /* gcFEATURE_BIT_NN_TF32_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SP_FP28_PRECISION */
+        0x0, /* gcFEATURE_BIT_NN_VIPSRAM_SHARE_AMONG_CLUSTERS */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_UNIQUE_ID */
+        0x0, /* gcFEATURE_BIT_NN_FAST_VIPSRAM */
+        0x0, /* gcFEATURE_BIT_NN_MOVE_DECOMPRESSOR_OUT_OF_CORE */
+        0x0, /* gcFEATURE_BIT_NN_SPLITKZ_REDUCE_SUM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FFD */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FP8_DOUBLE_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_DENORM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_POST_OP_FUSE */
+        0x0, /* gcFEATURE_BIT_T3D_SYNC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_GENERIC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_DOUBLE_CONV_OUT */
+        0x0, /* gcFEATURE_BIT_CMD_LOOP_STATE_LOAD_SAVE */
+        0x0, /* gcFEATURE_BIT_DYNAMIC_ACTIVATION_SCALE */
         0x0, /* gcFEATURE_BIT_NN_PER3DTILE_BUBBLE_FIX */
         0x0, /* gcFEATURE_BIT_NN_CACHELINE_MODE_PERF_FIX */
         0x0, /* gcFEATURE_BIT_NN_CONV1x1_PERF_FIX */
@@ -2772,6 +2837,10 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_B2B_RETRUN_NN_CMD_DONE_FIX */
         0x0, /* gcFEATURE_BIT_MULTI_AXI_ID_VLW_3D_TILE_INFO_FIX */
         0x0, /* gcFEATURE_BIT_TRSP2_SPECIAL_WORDSIZE_FIX */
+        0x0, /* gcFEATURE_BIT_FIX_PRE_CORE_LOTS_ZEROSKIP_KERNEL_FURTHER_FIX */
+        0x0, /* gcFEATURE_BIT_NN_2ND_IMAGE_TILE_LIMIT_1024B_FIX */
+        0x0, /* gcFEATURE_BIT_MULTIAXIID_TBUF_STALL_READY_FIX */
+        0x0, /* gcFEATURE_BIT_V940_TILESIZE_1024N_FIX */
         0x0, /* gcFEATURE_BIT_NN_INTERLEVE8 */
         0x0, /* gcFEATURE_BIT_NN_FP16_ALU */
         0x0, /* gcFEATURE_BIT_NN_INT16_ALU */
@@ -2835,6 +2904,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_PS_INSTRUCTION_COUNT */
         0x0, /* gcFEATURE_VALUE_PS_INPUT_COMPONENTS */
         0x0, /* gcFEATURE_VALUE_AIGM_MAX_SIZE */
+        0x0, /* gcFEATURE_VALUE_GPU_AXI_BUS_WIDTH */
+        0x0, /* gcFEATURE_VALUE_NumDPPUsPerCluster */
         0x40, /* gcFEATURE_VALUE_NNMadPerCore */
         0x2, /* gcFEATURE_VALUE_NNCoreCount */
         0x2, /* gcFEATURE_VALUE_NN_ACTIVE_CORE_COUNT */
@@ -2899,6 +2970,7 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_NN_COMMAND_BUFFER_SIZE */
         0x0, /* gcFEATURE_VALUE_DECOMP_VZ_GROUP_BITS */
         0x0, /* gcFEATURE_VALUE_NN_LOOP1_DP_NUMBER */
+        0x0, /* gcFEATURE_VALUE_NN_KERNEL_FIFO_IN_WIDTH */
         0x1, /* gcFEATURE_BIT_REG_FastClear */
         0x0, /* gcFEATURE_BIT_REG_SpecialAntiAliasing */
         0x1, /* gcFEATURE_BIT_REG_Pipe3D */
@@ -3332,6 +3404,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_DX11_SH_RCP_SQRT_PRECISSION */
         0x0, /* gcFEATURE_BIT_TILED_RESOURCE */
         0x0, /* gcFEATURE_BIT_SH_FP64 */
+        0x0, /* gcFEATURE_BIT_SH_INSTR_DENORM_SUPPORT */
+        0x0, /* gcFEATURE_BIT_SH_GA_SUPPORT */
         0x0, /* gcFEATURE_BIT_SH_SM6 */
         0x0, /* gcFEATURE_BIT_VGPU */
         0x0, /* gcFEATURE_BIT_DEC_NANO */
@@ -3533,7 +3607,6 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_BF16_I4_I8_QUANTIZATION */
         0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODE_SPLIT_STAGE */
         0x0, /* gcFEATURE_BIT_NN_POST_PROCESSOR_FL32 */
-        0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODER */
         0x0, /* gcFEATURE_BIT_NN_GROUP_QUANT_PHASE1 */
         0x0, /* gcFEATURE_BIT_NN_SH_IN_PARALLEL */
         0x0, /* gcFEATURE_BIT_NN_ASYNC_DMA */
@@ -3542,6 +3615,21 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_SP_ENHANCEMENT */
         0x0, /* gcFEATURE_BIT_NN_SUPPORT_SEPARATE_STREAMBUF_ADDR */
         0x0, /* gcFEATURE_BIT_NN_TF32_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SP_FP28_PRECISION */
+        0x0, /* gcFEATURE_BIT_NN_VIPSRAM_SHARE_AMONG_CLUSTERS */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_UNIQUE_ID */
+        0x0, /* gcFEATURE_BIT_NN_FAST_VIPSRAM */
+        0x0, /* gcFEATURE_BIT_NN_MOVE_DECOMPRESSOR_OUT_OF_CORE */
+        0x0, /* gcFEATURE_BIT_NN_SPLITKZ_REDUCE_SUM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FFD */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FP8_DOUBLE_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_DENORM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_POST_OP_FUSE */
+        0x0, /* gcFEATURE_BIT_T3D_SYNC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_GENERIC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_DOUBLE_CONV_OUT */
+        0x0, /* gcFEATURE_BIT_CMD_LOOP_STATE_LOAD_SAVE */
+        0x0, /* gcFEATURE_BIT_DYNAMIC_ACTIVATION_SCALE */
         0x1, /* gcFEATURE_BIT_NN_PER3DTILE_BUBBLE_FIX */
         0x1, /* gcFEATURE_BIT_NN_CACHELINE_MODE_PERF_FIX */
         0x1, /* gcFEATURE_BIT_NN_CONV1x1_PERF_FIX */
@@ -3680,6 +3768,10 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_B2B_RETRUN_NN_CMD_DONE_FIX */
         0x0, /* gcFEATURE_BIT_MULTI_AXI_ID_VLW_3D_TILE_INFO_FIX */
         0x0, /* gcFEATURE_BIT_TRSP2_SPECIAL_WORDSIZE_FIX */
+        0x0, /* gcFEATURE_BIT_FIX_PRE_CORE_LOTS_ZEROSKIP_KERNEL_FURTHER_FIX */
+        0x0, /* gcFEATURE_BIT_NN_2ND_IMAGE_TILE_LIMIT_1024B_FIX */
+        0x0, /* gcFEATURE_BIT_MULTIAXIID_TBUF_STALL_READY_FIX */
+        0x0, /* gcFEATURE_BIT_V940_TILESIZE_1024N_FIX */
         0x0, /* gcFEATURE_BIT_NN_INTERLEVE8 */
         0x1, /* gcFEATURE_BIT_NN_FP16_ALU */
         0x1, /* gcFEATURE_BIT_NN_INT16_ALU */
@@ -3743,6 +3835,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_PS_INSTRUCTION_COUNT */
         0x0, /* gcFEATURE_VALUE_PS_INPUT_COMPONENTS */
         0x0, /* gcFEATURE_VALUE_AIGM_MAX_SIZE */
+        0x0, /* gcFEATURE_VALUE_GPU_AXI_BUS_WIDTH */
+        0x0, /* gcFEATURE_VALUE_NumDPPUsPerCluster */
         0x40, /* gcFEATURE_VALUE_NNMadPerCore */
         0x2, /* gcFEATURE_VALUE_NNCoreCount */
         0x2, /* gcFEATURE_VALUE_NN_ACTIVE_CORE_COUNT */
@@ -3807,6 +3901,7 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_NN_COMMAND_BUFFER_SIZE */
         0x10, /* gcFEATURE_VALUE_DECOMP_VZ_GROUP_BITS */
         0x3, /* gcFEATURE_VALUE_NN_LOOP1_DP_NUMBER */
+        0x80, /* gcFEATURE_VALUE_NN_KERNEL_FIFO_IN_WIDTH */
         0x1, /* gcFEATURE_BIT_REG_FastClear */
         0x0, /* gcFEATURE_BIT_REG_SpecialAntiAliasing */
         0x1, /* gcFEATURE_BIT_REG_Pipe3D */
@@ -4240,6 +4335,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_DX11_SH_RCP_SQRT_PRECISSION */
         0x0, /* gcFEATURE_BIT_TILED_RESOURCE */
         0x0, /* gcFEATURE_BIT_SH_FP64 */
+        0x0, /* gcFEATURE_BIT_SH_INSTR_DENORM_SUPPORT */
+        0x0, /* gcFEATURE_BIT_SH_GA_SUPPORT */
         0x0, /* gcFEATURE_BIT_SH_SM6 */
         0x0, /* gcFEATURE_BIT_VGPU */
         0x0, /* gcFEATURE_BIT_DEC_NANO */
@@ -4441,7 +4538,6 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_BF16_I4_I8_QUANTIZATION */
         0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODE_SPLIT_STAGE */
         0x0, /* gcFEATURE_BIT_NN_POST_PROCESSOR_FL32 */
-        0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODER */
         0x0, /* gcFEATURE_BIT_NN_GROUP_QUANT_PHASE1 */
         0x0, /* gcFEATURE_BIT_NN_SH_IN_PARALLEL */
         0x0, /* gcFEATURE_BIT_NN_ASYNC_DMA */
@@ -4450,6 +4546,21 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_SP_ENHANCEMENT */
         0x0, /* gcFEATURE_BIT_NN_SUPPORT_SEPARATE_STREAMBUF_ADDR */
         0x0, /* gcFEATURE_BIT_NN_TF32_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SP_FP28_PRECISION */
+        0x0, /* gcFEATURE_BIT_NN_VIPSRAM_SHARE_AMONG_CLUSTERS */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_UNIQUE_ID */
+        0x0, /* gcFEATURE_BIT_NN_FAST_VIPSRAM */
+        0x0, /* gcFEATURE_BIT_NN_MOVE_DECOMPRESSOR_OUT_OF_CORE */
+        0x0, /* gcFEATURE_BIT_NN_SPLITKZ_REDUCE_SUM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FFD */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FP8_DOUBLE_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_DENORM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_POST_OP_FUSE */
+        0x0, /* gcFEATURE_BIT_T3D_SYNC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_GENERIC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_DOUBLE_CONV_OUT */
+        0x0, /* gcFEATURE_BIT_CMD_LOOP_STATE_LOAD_SAVE */
+        0x0, /* gcFEATURE_BIT_DYNAMIC_ACTIVATION_SCALE */
         0x0, /* gcFEATURE_BIT_NN_PER3DTILE_BUBBLE_FIX */
         0x0, /* gcFEATURE_BIT_NN_CACHELINE_MODE_PERF_FIX */
         0x1, /* gcFEATURE_BIT_NN_CONV1x1_PERF_FIX */
@@ -4588,6 +4699,10 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x1, /* gcFEATURE_BIT_B2B_RETRUN_NN_CMD_DONE_FIX */
         0x1, /* gcFEATURE_BIT_MULTI_AXI_ID_VLW_3D_TILE_INFO_FIX */
         0x1, /* gcFEATURE_BIT_TRSP2_SPECIAL_WORDSIZE_FIX */
+        0x1, /* gcFEATURE_BIT_FIX_PRE_CORE_LOTS_ZEROSKIP_KERNEL_FURTHER_FIX */
+        0x1, /* gcFEATURE_BIT_NN_2ND_IMAGE_TILE_LIMIT_1024B_FIX */
+        0x1, /* gcFEATURE_BIT_MULTIAXIID_TBUF_STALL_READY_FIX */
+        0x0, /* gcFEATURE_BIT_V940_TILESIZE_1024N_FIX */
         0x0, /* gcFEATURE_BIT_NN_INTERLEVE8 */
         0x0, /* gcFEATURE_BIT_NN_FP16_ALU */
         0x0, /* gcFEATURE_BIT_NN_INT16_ALU */
@@ -4651,6 +4766,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_PS_INSTRUCTION_COUNT */
         0x0, /* gcFEATURE_VALUE_PS_INPUT_COMPONENTS */
         0x0, /* gcFEATURE_VALUE_AIGM_MAX_SIZE */
+        0x0, /* gcFEATURE_VALUE_GPU_AXI_BUS_WIDTH */
+        0x0, /* gcFEATURE_VALUE_NumDPPUsPerCluster */
         0x40, /* gcFEATURE_VALUE_NNMadPerCore */
         0x2, /* gcFEATURE_VALUE_NNCoreCount */
         0x2, /* gcFEATURE_VALUE_NN_ACTIVE_CORE_COUNT */
@@ -4715,6 +4832,7 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_NN_COMMAND_BUFFER_SIZE */
         0x10, /* gcFEATURE_VALUE_DECOMP_VZ_GROUP_BITS */
         0x3, /* gcFEATURE_VALUE_NN_LOOP1_DP_NUMBER */
+        0x80, /* gcFEATURE_VALUE_NN_KERNEL_FIFO_IN_WIDTH */
         0x1, /* gcFEATURE_BIT_REG_FastClear */
         0x0, /* gcFEATURE_BIT_REG_SpecialAntiAliasing */
         0x1, /* gcFEATURE_BIT_REG_Pipe3D */
@@ -5148,6 +5266,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_DX11_SH_RCP_SQRT_PRECISSION */
         0x0, /* gcFEATURE_BIT_TILED_RESOURCE */
         0x0, /* gcFEATURE_BIT_SH_FP64 */
+        0x0, /* gcFEATURE_BIT_SH_INSTR_DENORM_SUPPORT */
+        0x0, /* gcFEATURE_BIT_SH_GA_SUPPORT */
         0x0, /* gcFEATURE_BIT_SH_SM6 */
         0x0, /* gcFEATURE_BIT_VGPU */
         0x0, /* gcFEATURE_BIT_DEC_NANO */
@@ -5349,7 +5469,6 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_BF16_I4_I8_QUANTIZATION */
         0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODE_SPLIT_STAGE */
         0x0, /* gcFEATURE_BIT_NN_POST_PROCESSOR_FL32 */
-        0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODER */
         0x0, /* gcFEATURE_BIT_NN_GROUP_QUANT_PHASE1 */
         0x0, /* gcFEATURE_BIT_NN_SH_IN_PARALLEL */
         0x0, /* gcFEATURE_BIT_NN_ASYNC_DMA */
@@ -5358,6 +5477,21 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_SP_ENHANCEMENT */
         0x0, /* gcFEATURE_BIT_NN_SUPPORT_SEPARATE_STREAMBUF_ADDR */
         0x0, /* gcFEATURE_BIT_NN_TF32_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SP_FP28_PRECISION */
+        0x0, /* gcFEATURE_BIT_NN_VIPSRAM_SHARE_AMONG_CLUSTERS */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_UNIQUE_ID */
+        0x0, /* gcFEATURE_BIT_NN_FAST_VIPSRAM */
+        0x0, /* gcFEATURE_BIT_NN_MOVE_DECOMPRESSOR_OUT_OF_CORE */
+        0x0, /* gcFEATURE_BIT_NN_SPLITKZ_REDUCE_SUM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FFD */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FP8_DOUBLE_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_DENORM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_POST_OP_FUSE */
+        0x0, /* gcFEATURE_BIT_T3D_SYNC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_GENERIC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_DOUBLE_CONV_OUT */
+        0x0, /* gcFEATURE_BIT_CMD_LOOP_STATE_LOAD_SAVE */
+        0x0, /* gcFEATURE_BIT_DYNAMIC_ACTIVATION_SCALE */
         0x0, /* gcFEATURE_BIT_NN_PER3DTILE_BUBBLE_FIX */
         0x0, /* gcFEATURE_BIT_NN_CACHELINE_MODE_PERF_FIX */
         0x1, /* gcFEATURE_BIT_NN_CONV1x1_PERF_FIX */
@@ -5496,6 +5630,10 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x1, /* gcFEATURE_BIT_B2B_RETRUN_NN_CMD_DONE_FIX */
         0x1, /* gcFEATURE_BIT_MULTI_AXI_ID_VLW_3D_TILE_INFO_FIX */
         0x1, /* gcFEATURE_BIT_TRSP2_SPECIAL_WORDSIZE_FIX */
+        0x1, /* gcFEATURE_BIT_FIX_PRE_CORE_LOTS_ZEROSKIP_KERNEL_FURTHER_FIX */
+        0x1, /* gcFEATURE_BIT_NN_2ND_IMAGE_TILE_LIMIT_1024B_FIX */
+        0x1, /* gcFEATURE_BIT_MULTIAXIID_TBUF_STALL_READY_FIX */
+        0x0, /* gcFEATURE_BIT_V940_TILESIZE_1024N_FIX */
         0x0, /* gcFEATURE_BIT_NN_INTERLEVE8 */
         0x0, /* gcFEATURE_BIT_NN_FP16_ALU */
         0x1, /* gcFEATURE_BIT_NN_INT16_ALU */
@@ -5559,6 +5697,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_PS_INSTRUCTION_COUNT */
         0x0, /* gcFEATURE_VALUE_PS_INPUT_COMPONENTS */
         0x0, /* gcFEATURE_VALUE_AIGM_MAX_SIZE */
+        0x0, /* gcFEATURE_VALUE_GPU_AXI_BUS_WIDTH */
+        0x0, /* gcFEATURE_VALUE_NumDPPUsPerCluster */
         0x40, /* gcFEATURE_VALUE_NNMadPerCore */
         0x2, /* gcFEATURE_VALUE_NNCoreCount */
         0x2, /* gcFEATURE_VALUE_NN_ACTIVE_CORE_COUNT */
@@ -5623,6 +5763,7 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_VALUE_NN_COMMAND_BUFFER_SIZE */
         0x10, /* gcFEATURE_VALUE_DECOMP_VZ_GROUP_BITS */
         0x3, /* gcFEATURE_VALUE_NN_LOOP1_DP_NUMBER */
+        0x80, /* gcFEATURE_VALUE_NN_KERNEL_FIFO_IN_WIDTH */
         0x1, /* gcFEATURE_BIT_REG_FastClear */
         0x0, /* gcFEATURE_BIT_REG_SpecialAntiAliasing */
         0x1, /* gcFEATURE_BIT_REG_Pipe3D */
@@ -6056,6 +6197,8 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_DX11_SH_RCP_SQRT_PRECISSION */
         0x0, /* gcFEATURE_BIT_TILED_RESOURCE */
         0x0, /* gcFEATURE_BIT_SH_FP64 */
+        0x0, /* gcFEATURE_BIT_SH_INSTR_DENORM_SUPPORT */
+        0x0, /* gcFEATURE_BIT_SH_GA_SUPPORT */
         0x0, /* gcFEATURE_BIT_SH_SM6 */
         0x0, /* gcFEATURE_BIT_VGPU */
         0x0, /* gcFEATURE_BIT_DEC_NANO */
@@ -6257,7 +6400,6 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_BF16_I4_I8_QUANTIZATION */
         0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODE_SPLIT_STAGE */
         0x0, /* gcFEATURE_BIT_NN_POST_PROCESSOR_FL32 */
-        0x0, /* gcFEATURE_BIT_NN_HIGH_PERF_DECODER */
         0x0, /* gcFEATURE_BIT_NN_GROUP_QUANT_PHASE1 */
         0x0, /* gcFEATURE_BIT_NN_SH_IN_PARALLEL */
         0x0, /* gcFEATURE_BIT_NN_ASYNC_DMA */
@@ -6266,6 +6408,21 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x0, /* gcFEATURE_BIT_NN_SP_ENHANCEMENT */
         0x0, /* gcFEATURE_BIT_NN_SUPPORT_SEPARATE_STREAMBUF_ADDR */
         0x0, /* gcFEATURE_BIT_NN_TF32_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SP_FP28_PRECISION */
+        0x0, /* gcFEATURE_BIT_NN_VIPSRAM_SHARE_AMONG_CLUSTERS */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_UNIQUE_ID */
+        0x0, /* gcFEATURE_BIT_NN_FAST_VIPSRAM */
+        0x0, /* gcFEATURE_BIT_NN_MOVE_DECOMPRESSOR_OUT_OF_CORE */
+        0x0, /* gcFEATURE_BIT_NN_SPLITKZ_REDUCE_SUM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FFD */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_FP8_DOUBLE_MAC */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_DENORM */
+        0x0, /* gcFEATURE_BIT_NN_SUPPORT_POST_OP_FUSE */
+        0x0, /* gcFEATURE_BIT_T3D_SYNC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_GENERIC_DMA */
+        0x0, /* gcFEATURE_BIT_NN_DOUBLE_CONV_OUT */
+        0x0, /* gcFEATURE_BIT_CMD_LOOP_STATE_LOAD_SAVE */
+        0x0, /* gcFEATURE_BIT_DYNAMIC_ACTIVATION_SCALE */
         0x0, /* gcFEATURE_BIT_NN_PER3DTILE_BUBBLE_FIX */
         0x0, /* gcFEATURE_BIT_NN_CACHELINE_MODE_PERF_FIX */
         0x1, /* gcFEATURE_BIT_NN_CONV1x1_PERF_FIX */
@@ -6404,6 +6561,10 @@ static gcsFEATURE_DATABASE gChipInfo[] = {
         0x1, /* gcFEATURE_BIT_B2B_RETRUN_NN_CMD_DONE_FIX */
         0x1, /* gcFEATURE_BIT_MULTI_AXI_ID_VLW_3D_TILE_INFO_FIX */
         0x1, /* gcFEATURE_BIT_TRSP2_SPECIAL_WORDSIZE_FIX */
+        0x1, /* gcFEATURE_BIT_FIX_PRE_CORE_LOTS_ZEROSKIP_KERNEL_FURTHER_FIX */
+        0x1, /* gcFEATURE_BIT_NN_2ND_IMAGE_TILE_LIMIT_1024B_FIX */
+        0x1, /* gcFEATURE_BIT_MULTIAXIID_TBUF_STALL_READY_FIX */
+        0x0, /* gcFEATURE_BIT_V940_TILESIZE_1024N_FIX */
         0x0, /* gcFEATURE_BIT_NN_INTERLEVE8 */
         0x0, /* gcFEATURE_BIT_NN_FP16_ALU */
         0x1, /* gcFEATURE_BIT_NN_INT16_ALU */
-- 
2.51.0

